test/simulator/src/exec_callee.rs

```diff
@@ -38,15 +38,16 @@ fn main() {
     let args = env::args_os()
         .into_iter()
         .map(|arg| CString::new(arg.into_vec()).unwrap())
-        .collect::<Vec<_>>();
-    let argc = args.len() as u64;
-    let mut argv = args
+        .collect::<Vec<_>>()
+        .leak();
+    let argv = args
         .iter()
-        .map(|cstring| cstring.to_bytes_with_nul().as_ptr())
-        .collect::<Vec<_>>();
-    argv.push(std::ptr::null());
```

Comment:
> A null pointer always follows the last element: argv[argc] is this null pointer.
https://www.gnu.org/software/libc/manual/html_node/Program-Arguments.html

Reply:
Other changes beside this is LGTM.

Reply:
`set_argv` expects `&[&CStr]`, whose length is equal to `argc`. Besides there is no such thing as a null reference.

Reply:
Rust is rust, program protocol is program protocol, why we here parse `argv` as `CStr` instead of `String`?

Reply:
And what if the callee contract is written in other language (such as C99) and it expect a null pointer at the and of `argv`?

Reply:
Not all `&CStr` can be converted to `&str` or `String`.

The contract for the argv API is on a higher level than the argv pointer, and it does not include the final null pointer. There's a nice symmetry between `exec` and `argv`. The `argv` argument of `exec` is also `&[&CStr]` and it also does not contain the last null pointer.

A callee contract written in other language wouldn't be tested with `set_argv`, so it does not seem to be relevant.

Reply:
Sorry, I'm not aware of current is a callee contract.

If it's a caller contract, should insert null pointer at the end, and it's already handled in [`syscalls::exec()`](https://github.com/nervosnetwork/ckb-std/blob/01a6b1bb9593760a9b956f3a12764f83492c9a0a/src/syscalls/native.rs#L547) already.